discord/embeds.py

```diff
@@ -123,20 +65,32 @@ class EmbedAuthor(EmbedProxy):
     def __init__(
         self,
         name: str,
-        url: MaybeEmpty[str] = EmptyEmbed,
-        icon_url: MaybeEmpty[str] = EmptyEmbed,
-        proxy_icon_url: MaybeEmpty[str] = EmptyEmbed,
+        url: str | None = None,
+        icon_url: str | None = None,
     ) -> None:
-        layer = {
-            k: v
-            for k, v in locals().items()
-            if k in {"name", "url", "icon_url", "proxy_icon_url"}
-            and v is not EmptyEmbed
+        self.name: str = name
+        self.url: str | None = url
+        self.icon_url: str | None = icon_url
+        self.proxy_icon_url: str | None = None
```

Comment:
The old `not` check was necessary to ensure `None` never ended up in the payload.
```py
discord.Embed(author=discord.EmbedAuthor(name="Nelo"))
```
Raises when sent
```py
discord.errors.HTTPException: 400 Bad Request (error code: 50035): Invalid Form Body
In embeds.0.author.icon_url: Not a well formed URL.
In embeds.0.author.url: Scheme "none" is not supported. Scheme must be one of ('http', 'https'). Not a well formed URL.
```
Please recreate the previous logic as you see fit.

Reply:
Ah that's right. Working on it